### PR TITLE
added yara_binary_search docs

### PR DESCRIPTION
**Pull request by jstucke**

### CODE REVIEW

File: src/helperFunctions/yara_binary_search.py [hunk 1]
```diff
@@ -75,25 +85,37 @@ def get_binary_search_result(self, task: Tuple[bytes, Optional[str]]):
...
+    '''
+    Get the exception that is caused by trying to compile ``rules_file`` with yara or ``None`` if there is none.
 
-def get_yara_error(rules_file):
+    :param rules_file: A string containing yara rules.
+    :result: The exception if compiling the rules causes an exception or ``None`` otherwise.
+    '''
     if isinstance(rules_file, bytes):
         rules_file = rules_file.decode()
     try:
```
Comment (dorpvom) [inline src/helperFunctions/yara_binary_search.py:121]: Is there a specific reason why you haven't documented `class YaraBinarySearchScannerDbInterface` ?

File: src/helperFunctions/yara_binary_search.py [hunk 2]
```diff
@@ -12,23 +13,30 @@
 
 
 class YaraBinarySearchScanner:
+    '''
+    A scanner for searching with YARA rules on the binaries belonging to the firmware and file objects contained
```
Comment (dorpvom) [inline src/helperFunctions/yara_binary_search.py:17]: It think we can improve the wording here. How 'bout:

This class provides functionality to scan files in the database for [yara](http://virustotal.github.io/yara/) patterns. The public method allows to either match a given set of patterns on all files in the database or focus only on files included in a single firmware.

**Review by dorpvom — CHANGES_REQUESTED** [review]

File: src/helperFunctions/yara_binary_search.py (hunk 1 above)
Comment (jstucke) [reply src/helperFunctions/yara_binary_search.py:121]: Yes: it is only used internally by the other functions (I could also have renamed it to `_YaraBinarySearchScannerDbInterface` but that looked ugly)

File: src/helperFunctions/yara_binary_search.py (hunk 2 above)
Comment (jstucke) [reply src/helperFunctions/yara_binary_search.py:17]: I agree → changed

File: src/helperFunctions/yara_binary_search.py (hunk 1 above)
Comment (dorpvom) [reply src/helperFunctions/yara_binary_search.py:122]: Haha. Yes it does

**Review by dorpvom — APPROVED** [review]

**Pull request merged**